### PR TITLE
lib name specifying + proper scoping + multifile tests

### PR DESCRIPTION
**Pull request by rooooooooob**
# --lib-name arg to change exported library names 

Also changes format to be the rust/wasm instead of core/wasm in line
with current CML format.

Includes minor fixes for when wrapper structs are used within (this was
broken when `AsRef` was introduced instead of a pub(crate) tuple field)
a different WASM file.

# Proper scoping/imports when using from other files

Both crates: proper imports referring to used types rather than blind
`use foo::*;`'s everywhere.

WASM crate refers to proper scoping directly as https://github.com/dcSpark/cddl-codegen/pull/128 used non-pub
imports in the rust crate for other files which broke multi-file support
in WASM as they referred to `wasm::Foo` instead of e.g.
`wasm::foo::Foo`.

Also changed alias name for arrays to `FooList` instead of `Foos` as
it's more explicit and works better with combining.

# Multi-file test added

This will help catch things that would break multifile support accidentally

### CODE REVIEW

File: src/intermediate.rs [hunk 1]
```diff
@@ -11,6 +11,23 @@ use crate::utils::{
     is_identifier_user_defined,
 };
 
+#[derive(Debug)]
```
Comment (rooooooooob) [inline src/intermediate.rs:14]: I had originally added an extra field and for that made it its own struct but later deleted that field. I still think it's worth it to keep the change anyway as it's more readable vs a tuple.

File: src/generation.rs [hunk 2]
```diff
@@ -846,17 +876,59 @@ impl GenerationScope {
...
+                            );
+                        } else if idents.len() == 1 {
+                            content.push_import(
+                                import_scope,
+                                idents.first().unwrap().to_string(),
+                                None,
+                            );
+                        }
+                    }
+                }
```
Comment (SebastienGllmt) [inline src/generation.rs:916]: Seems like a lot of code duplication

File: src/intermediate.rs [hunk 3]
```diff
@@ -79,20 +95,130 @@ impl<'a> IntermediateTypes<'a> {
...
+                        set_ref(refs, types, current_scope, rust_ident);
+                    }
+                }
+                ConceptualRustType::Rust(rust_ident) => {
+                    set_ref(refs, types, current_scope, rust_ident)
+                }
+                ConceptualRustType::Array(elem_ty) => {
+                    if wasm && !elem_ty.directly_wasm_exposable() {
+                        // TODO: we should be doing array wrappers where they are declared or used
+                        // but for the latter, what to do if multiple places use it? default to lib?
```
Comment (SebastienGllmt) [inline src/intermediate.rs:148]: For the TODO here and below, it may be good to create a Github issue for this and reference it in the code

**Review by SebastienGllmt — APPROVED** [review]
LGTM other than nit-picky stuff

File: src/intermediate.rs (hunk 3 above)
Comment (rooooooooob) [reply src/intermediate.rs:148]: Will fix in next PR as this PR already contains a lot of different stuff

File: src/generation.rs (hunk 2 above)
Comment (rooooooooob) [reply src/generation.rs:916]: Will fix in next PR as this PR already contains a lot of different stuff

**Pull request merged**